sdk/rust-language-support/examples/image-processing/src/main.rs

```diff
@@ -0,0 +1,70 @@
+//! A simple linear-regression example
+//!
+//! ## Context
+//!
+//! A relatively-low powered device (think e.g. an M-class microcontroller, or similar)
+//! wants to offload some heavy, complex calculation to a third-party---here represented by a linear
+//! regression problem.  The owner of the device wishes to ensure that the computation was
+//! faithfully executed---that is, the results are trustworthy and are definitely the output of a
+//! linear regression algorithm---and also are kept secret from third parties wishing to see what is
+//! happening on the device.
+//!
+//! Inputs:                  1.
+//! Assumed form of inputs:  one Pinecode-encoded `Vec<(f64, f64)>` representing a dataset of (x, y)
+//!                          co-ordinates from which a linear relationship is to be extracted.
+//! Ensured form of outputs: A Pinecode-encoded `LinearRegression` struct (see below) consisting of
+//!                          a gradient and a Y-intercept, representing the best linear fit for the
+//!                          input data.
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Licensing and copyright notice
+//!
+//! See the `LICENSE.markdown` file in the Veracruz root directory for
+//! information on licensing and copyright.
+
+extern crate image;
+
+use image::{GenericImageView, imageops, ImageFormat, io::Reader};
+use std::process::exit;
+use wasi_types::ErrNo;
+
+/// Reads two input sources from the Veracruz host: the first input source is assumed to be a vector
```

Comment:
This `rustdoc` comment is incorrect.  It needs replacing with something specific to this code.